### PR TITLE
Make the generated opam files release-ready

### PR DESCRIPTION
**Pull request by mjambon**
This should eliminate discrepancies between the opam files that are derived from `dune-project` and the opam files that are acceptable for the opam-repository. This is meant to avoid some manual editing on the opam-repository pull request.

### PR checklist

- [x] New code has tests to catch future regressions
- [x] Documentation is up-to-date
- [x] `CHANGES.md` is up-to-date

### CODE REVIEW

**Comment by mjambon** [comment]
Here's the diff between the generated opam files and the ones used for release 2.3.3. I think it's all good now. All the differences I see are expected or not real differences.
```
~/atd $ for x in atd atds atdj atdpy atdgen atdgen-codec-runtime atdgen-runtime; do diff -u $x.opam ~/opam-repository/packages/$x/$x.2.3.3/opam; done
--- atd.opam	2022-03-17 18:15:35.081495794 -0700
+++ /home/martin/opam-repository/packages/atd/atd.2.3.3/opam	2022-03-16 21:30:51.071481714 -0700
@@ -1,4 +1,3 @@
-# This file is generated by dune, edit dune-project instead
 opam-version: "2.0"
 synopsis: "Parser for the ATD data format description language"
 description: """
@@ -13,60 +12,26 @@
 ]
 authors: [
   "Martin Jambon <martin@mjambon.com>"
-  "Rudi Grinberg <rudi.grinberg@gmail.com>"
-  "Martin Jambon <martin@r2c.dev>"
-  "Martin Jambon <github@mjambon.com>"
-  "Ivan Jager <aij+git@mrph.org>"
-  "oleksiy <oleksiy.golovko@ahrefs.com>"
   "David Sheets <sheets@alum.mit.edu>"
   "Rudi Grinberg <me@rgrinberg.com>"
-  "Martin Jambon <martin@esper.com>"
+  "Ivan Jager <aij+git@mrph.org>"
   "Jeff Meister <nanaki@gmail.com>"
-  "Caio Wakamatsu <caio.wakamatsu@ahrefs.com>"
   "Carmelo Piccione <carmelo.piccione@gmail.com>"
-  "Daniel Weil <danweil68@gmail.com>"
-  "Egor Chemokhonenko <egor.chemohonenko@ahrefs.com>"
-  "Gabriel Scherer <gabriel.scherer@gmail.com>"
   "Raman Varabets <roman.vorobets@gmail.com>"
-  "tzm <frank@boldsolutions.de>"
   "Mathieu Baudet <mathieubaudet@fb.com>"
-  "Oleksiy Golovko <alexei.golovko@gmail.com>"
   "Rauan Mayemir <rauan@mayemir.io>"
-  "Carmelo Piccione <cep1@solvuu.com>"
-  "John Billings <john@monkeynut.org>"
   "Louis Roché <louis@louisroche.net>"
   "Brendan Long <self@brendanlong.com>"
-  "Chris Yocum <cyocum@gmail.com>"
-  "Louis Roché (Ahrefs) <louis.roche@ahrefs.com>"
-  "Louis Roché <louis.roche@ahrefs.com>"
-  "Pavel Antoshkin <pavel.antoshkin@ahrefs.com>"
-  "Pierre Boutillier <pierre.boutillier@laposte.net>"
-  "Shon Feder <shon.feder@key.me>"
-  "Anurag Soni <anuragsoni.13@gmail.com>"
-  "Arjun Ravi Narayan <arjunravinarayan@gmail.com>"
-  "Asya-kawai <kawai-toshiki@aintek.xyz>"
   "Christophe Troestler <christophe.Troestler@umons.ac.be>"
-  "Damien Doligez <ddoligez@janestreet.com>"
-  "Daniel M <dan.mntg@gmail.com>"
-  "Ding Xiang Fei <dingxiangfei2009@protonmail.ch>"
-  "François Pottier <francois.pottier@inria.fr>"
-  "Javier Chavarri <javier.chavarri@gmail.com>"
-  "Kate <kit.ty.kate@disroot.org>"
-  "Louis <louis.roche@ahrefs.com>"
-  "Louis Roché <louis@cryptosense.com>"
-  "Raman Varabets <raman+git@ahrefs.com>"
-  "Stephane Legrand <slegrand45@gmail.com>"
   "Vincent Bernardoff <vb@luminar.eu.org>"
   "haoyang <haoyang@esper.co>"
-  "pmundkur <prashanth.mundkur@gmail.com>"
-  "ygrek <ygrek@autistici.org>"
 ]
 license: "MIT"
 homepage: "https://github.com/ahrefs/atd"
 bug-reports: "https://github.com/ahrefs/atd/issues"
 depends: [
-  "dune" {>= "2.8"}
   "ocaml" {>= "4.08"}
+  "dune" {>= "2.8"}
   "menhir" {>= "20180523" & < "20211215"}
   "easy-format"
   "alcotest" {with-test}
@@ -87,3 +52,12 @@
     "@doc" {with-doc}
   ]
 ]
+url {
+  src:
+    "https://github.com/ahrefs/atd/releases/download/2.3.3/atdgen-codec-runtime-2.3.3.tbz"
+  checksum: [
+    "sha256=3a25addaebfe13b48a44fe3e6de88e9c50ea62433dfd3684f0a894d957d1eb47"
+    "sha512=a4375cb465b9d34b1bab49f331600b108241342e1957d9a2332eec5dd55ecf70021c21a6bb2431f015359e46dffbdae06a21c4f5c98c7e9ef3911bb7dca9461b"
+  ]
+}
+x-commit-hash: "28b612fbe750408ce58c3e952196460b2d7f5894"
--- atds.opam	2022-03-17 18:15:35.081495794 -0700
+++ /home/martin/opam-repository/packages/atds/atds.2.3.3/opam	2022-03-16 21:30:51.083481686 -0700
@@ -1,4 +1,3 @@
-# This file is generated by dune, edit dune-project instead
 opam-version: "2.0"
 synopsis: "ATD Code generator for Scala"
 description: "ATD Code generator for Scala"
@@ -9,60 +8,26 @@
 ]
 authors: [
   "Martin Jambon <martin@mjambon.com>"
-  "Rudi Grinberg <rudi.grinberg@gmail.com>"
-  "Martin Jambon <martin@r2c.dev>"
-  "Martin Jambon <github@mjambon.com>"
-  "Ivan Jager <aij+git@mrph.org>"
-  "oleksiy <oleksiy.golovko@ahrefs.com>"
   "David Sheets <sheets@alum.mit.edu>"
   "Rudi Grinberg <me@rgrinberg.com>"
-  "Martin Jambon <martin@esper.com>"
+  "Ivan Jager <aij+git@mrph.org>"
   "Jeff Meister <nanaki@gmail.com>"
-  "Caio Wakamatsu <caio.wakamatsu@ahrefs.com>"
   "Carmelo Piccione <carmelo.piccione@gmail.com>"
-  "Daniel Weil <danweil68@gmail.com>"
-  "Egor Chemokhonenko <egor.chemohonenko@ahrefs.com>"
-  "Gabriel Scherer <gabriel.scherer@gmail.com>"
   "Raman Varabets <roman.vorobets@gmail.com>"
-  "tzm <frank@boldsolutions.de>"
   "Mathieu Baudet <mathieubaudet@fb.com>"
-  "Oleksiy Golovko <alexei.golovko@gmail.com>"
   "Rauan Mayemir <rauan@mayemir.io>"
-  "Carmelo Piccione <cep1@solvuu.com>"
-  "John Billings <john@monkeynut.org>"
   "Louis Roché <louis@louisroche.net>"
   "Brendan Long <self@brendanlong.com>"
-  "Chris Yocum <cyocum@gmail.com>"
-  "Louis Roché (Ahrefs) <louis.roche@ahrefs.com>"
-  "Louis Roché <louis.roche@ahrefs.com>"
-  "Pavel Antoshkin <pavel.antoshkin@ahrefs.com>"
-  "Pierre Boutillier <pierre.boutillier@laposte.net>"
-  "Shon Feder <shon.feder@key.me>"
-  "Anurag Soni <anuragsoni.13@gmail.com>"
-  "Arjun Ravi Narayan <arjunravinarayan@gmail.com>"
-  "Asya-kawai <kawai-toshiki@aintek.xyz>"
   "Christophe Troestler <christophe.Troestler@umons.ac.be>"
-  "Damien Doligez <ddoligez@janestreet.com>"
-  "Daniel M <dan.mntg@gmail.com>"
-  "Ding Xiang Fei <dingxiangfei2009@protonmail.ch>"
-  "François Pottier <francois.pottier@inria.fr>"
-  "Javier Chavarri <javier.chavarri@gmail.com>"
-  "Kate <kit.ty.kate@disroot.org>"
-  "Louis <louis.roche@ahrefs.com>"
-  "Louis Roché <louis@cryptosense.com>"
-  "Raman Varabets <raman+git@ahrefs.com>"
-  "Stephane Legrand <slegrand45@gmail.com>"
   "Vincent Bernardoff <vb@luminar.eu.org>"
   "haoyang <haoyang@esper.co>"
-  "pmundkur <prashanth.mundkur@gmail.com>"
-  "ygrek <ygrek@autistici.org>"
 ]
 license: "MIT"
 homepage: "https://github.com/ahrefs/atd"
 bug-reports: "https://github.com/ahrefs/atd/issues"
 depends: [
-  "dune" {>= "2.8"}
   "ocaml" {>= "4.08"}
+  "dune" {>= "2.8"}
   "atd" {>= "2.3.0"}
   "odoc" {with-doc}
 ]
@@ -80,3 +45,12 @@
     "@doc" {with-doc}
   ]
 ]
+url {
+  src:
+    "https://github.com/ahrefs/atd/releases/download/2.3.3/atdgen-codec-runtime-2.3.3.tbz"
+  checksum: [
+    "sha256=3a25addaebfe13b48a44fe3e6de88e9c50ea62433dfd3684f0a894d957d1eb47"
+    "sha512=a4375cb465b9d34b1bab49f331600b108241342e1957d9a2332eec5dd55ecf70021c21a6bb2431f015359e46dffbdae06a21c4f5c98c7e9ef3911bb7dca9461b"
+  ]
+}
+x-commit-hash: "28b612fbe750408ce58c3e952196460b2d7f5894"
--- atdj.opam	2022-03-17 18:15:35.081495794 -0700
+++ /home/martin/opam-repository/packages/atdj/atdj.2.3.3/opam	2022-03-16 21:30:51.079481696 -0700
@@ -1,4 +1,3 @@
-# This file is generated by dune, edit dune-project instead
 opam-version: "2.0"
 synopsis: "Java code generation for ATD"
 description: """
@@ -23,60 +22,26 @@
 ]
 authors: [
   "Martin Jambon <martin@mjambon.com>"
-  "Rudi Grinberg <rudi.grinberg@gmail.com>"
-  "Martin Jambon <martin@r2c.dev>"
-  "Martin Jambon <github@mjambon.com>"
-  "Ivan Jager <aij+git@mrph.org>"
-  "oleksiy <oleksiy.golovko@ahrefs.com>"
   "David Sheets <sheets@alum.mit.edu>"
   "Rudi Grinberg <me@rgrinberg.com>"
-  "Martin Jambon <martin@esper.com>"
+  "Ivan Jager <aij+git@mrph.org>"
   "Jeff Meister <nanaki@gmail.com>"
-  "Caio Wakamatsu <caio.wakamatsu@ahrefs.com>"
   "Carmelo Piccione <carmelo.piccione@gmail.com>"
-  "Daniel Weil <danweil68@gmail.com>"
-  "Egor Chemokhonenko <egor.chemohonenko@ahrefs.com>"
-  "Gabriel Scherer <gabriel.scherer@gmail.com>"
   "Raman Varabets <roman.vorobets@gmail.com>"
-  "tzm <frank@boldsolutions.de>"
   "Mathieu Baudet <mathieubaudet@fb.com>"
-  "Oleksiy Golovko <alexei.golovko@gmail.com>"
   "Rauan Mayemir <rauan@mayemir.io>"
-  "Carmelo Piccione <cep1@solvuu.com>"
-  "John Billings <john@monkeynut.org>"
   "Louis Roché <louis@louisroche.net>"
   "Brendan Long <self@brendanlong.com>"
-  "Chris Yocum <cyocum@gmail.com>"
-  "Louis Roché (Ahrefs) <louis.roche@ahrefs.com>"
-  "Louis Roché <louis.roche@ahrefs.com>"
-  "Pavel Antoshkin <pavel.antoshkin@ahrefs.com>"
-  "Pierre Boutillier <pierre.boutillier@laposte.net>"
-  "Shon Feder <shon.feder@key.me>"
-  "Anurag Soni <anuragsoni.13@gmail.com>"
-  "Arjun Ravi Narayan <arjunravinarayan@gmail.com>"
-  "Asya-kawai <kawai-toshiki@aintek.xyz>"
   "Christophe Troestler <christophe.Troestler@umons.ac.be>"
-  "Damien Doligez <ddoligez@janestreet.com>"
-  "Daniel M <dan.mntg@gmail.com>"
-  "Ding Xiang Fei <dingxiangfei2009@protonmail.ch>"
-  "François Pottier <francois.pottier@inria.fr>"
-  "Javier Chavarri <javier.chavarri@gmail.com>"
-  "Kate <kit.ty.kate@disroot.org>"
-  "Louis <louis.roche@ahrefs.com>"
-  "Louis Roché <louis@cryptosense.com>"
-  "Raman Varabets <raman+git@ahrefs.com>"
-  "Stephane Legrand <slegrand45@gmail.com>"
   "Vincent Bernardoff <vb@luminar.eu.org>"
   "haoyang <haoyang@esper.co>"
-  "pmundkur <prashanth.mundkur@gmail.com>"
-  "ygrek <ygrek@autistici.org>"
 ]
 license: "MIT"
 homepage: "https://github.com/ahrefs/atd"
 bug-reports: "https://github.com/ahrefs/atd/issues"
 depends: [
-  "dune" {>= "2.8"}
   "ocaml" {>= "4.08"}
+  "dune" {>= "2.8"}
   "atd" {>= "2.3.0"}
   "re"
   "odoc" {with-doc}
@@ -95,3 +60,12 @@
     "@doc" {with-doc}
   ]
 ]
+url {
+  src:
+    "https://github.com/ahrefs/atd/releases/download/2.3.3/atdgen-codec-runtime-2.3.3.tbz"
+  checksum: [
+    "sha256=3a25addaebfe13b48a44fe3e6de88e9c50ea62433dfd3684f0a894d957d1eb47"
+    "sha512=a4375cb465b9d34b1bab49f331600b108241342e1957d9a2332eec5dd55ecf70021c21a6bb2431f015359e46dffbdae06a21c4f5c98c7e9ef3911bb7dca9461b"
+  ]
+}
+x-commit-hash: "28b612fbe750408ce58c3e952196460b2d7f5894"
--- atdpy.opam	2022-03-17 18:15:35.081495794 -0700
+++ /home/martin/opam-repository/packages/atdpy/atdpy.2.3.3/opam	2022-03-16 21:30:51.079481696 -0700
@@ -1,4 +1,3 @@
-# This file is generated by dune, edit dune-project instead
 opam-version: "2.0"
 synopsis: "Python/mypy code generation for ATD APIs"
 description: "Python/mypy code generation for ATD APIs"
@@ -9,65 +8,30 @@
 ]
 authors: [
   "Martin Jambon <martin@mjambon.com>"
-  "Rudi Grinberg <rudi.grinberg@gmail.com>"
-  "Martin Jambon <martin@r2c.dev>"
-  "Martin Jambon <github@mjambon.com>"
-  "Ivan Jager <aij+git@mrph.org>"
-  "oleksiy <oleksiy.golovko@ahrefs.com>"
   "David Sheets <sheets@alum.mit.edu>"
   "Rudi Grinberg <me@rgrinberg.com>"
-  "Martin Jambon <martin@esper.com>"
+  "Ivan Jager <aij+git@mrph.org>"
   "Jeff Meister <nanaki@gmail.com>"
-  "Caio Wakamatsu <caio.wakamatsu@ahrefs.com>"
   "Carmelo Piccione <carmelo.piccione@gmail.com>"
-  "Daniel Weil <danweil68@gmail.com>"
-  "Egor Chemokhonenko <egor.chemohonenko@ahrefs.com>"
-  "Gabriel Scherer <gabriel.scherer@gmail.com>"
   "Raman Varabets <roman.vorobets@gmail.com>"
-  "tzm <frank@boldsolutions.de>"
   "Mathieu Baudet <mathieubaudet@fb.com>"
-  "Oleksiy Golovko <alexei.golovko@gmail.com>"
   "Rauan Mayemir <rauan@mayemir.io>"
-  "Carmelo Piccione <cep1@solvuu.com>"
-  "John Billings <john@monkeynut.org>"
   "Louis Roché <louis@louisroche.net>"
   "Brendan Long <self@brendanlong.com>"
-  "Chris Yocum <cyocum@gmail.com>"
-  "Louis Roché (Ahrefs) <louis.roche@ahrefs.com>"
-  "Louis Roché <louis.roche@ahrefs.com>"
-  "Pavel Antoshkin <pavel.antoshkin@ahrefs.com>"
-  "Pierre Boutillier <pierre.boutillier@laposte.net>"
-  "Shon Feder <shon.feder@key.me>"
-  "Anurag Soni <anuragsoni.13@gmail.com>"
-  "Arjun Ravi Narayan <arjunravinarayan@gmail.com>"
-  "Asya-kawai <kawai-toshiki@aintek.xyz>"
   "Christophe Troestler <christophe.Troestler@umons.ac.be>"
-  "Damien Doligez <ddoligez@janestreet.com>"
-  "Daniel M <dan.mntg@gmail.com>"
-  "Ding Xiang Fei <dingxiangfei2009@protonmail.ch>"
-  "François Pottier <francois.pottier@inria.fr>"
-  "Javier Chavarri <javier.chavarri@gmail.com>"
-  "Kate <kit.ty.kate@disroot.org>"
-  "Louis <louis.roche@ahrefs.com>"
-  "Louis Roché <louis@cryptosense.com>"
-  "Raman Varabets <raman+git@ahrefs.com>"
-  "Stephane Legrand <slegrand45@gmail.com>"
   "Vincent Bernardoff <vb@luminar.eu.org>"
   "haoyang <haoyang@esper.co>"
-  "pmundkur <prashanth.mundkur@gmail.com>"
-  "ygrek <ygrek@autistici.org>"
 ]
 license: "MIT"
 homepage: "https://github.com/ahrefs/atd"
 bug-reports: "https://github.com/ahrefs/atd/issues"
 depends: [
-  "dune" {>= "2.8"}
   "ocaml" {>= "4.08"}
+  "dune" {>= "2.8"}
+  "alcotest" {with-test}
   "atd" {>= "2.3.0"}
   "atdgen" {>= "2.3.0"}
   "cmdliner" {>= "1.1.0"}
-  "alcotest" {with-test}
-  "conf-python-3" {with-test}
   "odoc" {with-doc}
 ]
 build: [
@@ -80,8 +44,16 @@
     "-j"
     jobs
     "@install"
-    "@runtest" {with-test}
     "@doc" {with-doc}
   ]
 ]
 dev-repo: "git+https://github.com/ahrefs/atd.git"
+url {
+  src:
+    "https://github.com/ahrefs/atd/releases/download/2.3.3/atdgen-codec-runtime-2.3.3.tbz"
+  checksum: [
+    "sha256=3a25addaebfe13b48a44fe3e6de88e9c50ea62433dfd3684f0a894d957d1eb47"
+    "sha512=a4375cb465b9d34b1bab49f331600b108241342e1957d9a2332eec5dd55ecf70021c21a6bb2431f015359e46dffbdae06a21c4f5c98c7e9ef3911bb7dca9461b"
+  ]
+}
+x-commit-hash: "28b612fbe750408ce58c3e952196460b2d7f5894"
--- atdgen.opam	2022-03-17 18:15:35.081495794 -0700
+++ /home/martin/opam-repository/packages/atdgen/atdgen.2.3.3/opam	2022-03-16 21:30:51.071481714 -0700
@@ -1,4 +1,3 @@
-# This file is generated by dune, edit dune-project instead
 opam-version: "2.0"
 synopsis:
   "Generates efficient JSON serializers, deserializers and validators"
@@ -19,60 +18,26 @@
 ]
 authors: [
   "Martin Jambon <martin@mjambon.com>"
-  "Rudi Grinberg <rudi.grinberg@gmail.com>"
-  "Martin Jambon <martin@r2c.dev>"
-  "Martin Jambon <github@mjambon.com>"
-  "Ivan Jager <aij+git@mrph.org>"
-  "oleksiy <oleksiy.golovko@ahrefs.com>"
   "David Sheets <sheets@alum.mit.edu>"
   "Rudi Grinberg <me@rgrinberg.com>"
-  "Martin Jambon <martin@esper.com>"
+  "Ivan Jager <aij+git@mrph.org>"
   "Jeff Meister <nanaki@gmail.com>"
-  "Caio Wakamatsu <caio.wakamatsu@ahrefs.com>"
   "Carmelo Piccione <carmelo.piccione@gmail.com>"
-  "Daniel Weil <danweil68@gmail.com>"
-  "Egor Chemokhonenko <egor.chemohonenko@ahrefs.com>"
-  "Gabriel Scherer <gabriel.scherer@gmail.com>"
   "Raman Varabets <roman.vorobets@gmail.com>"
-  "tzm <frank@boldsolutions.de>"
   "Mathieu Baudet <mathieubaudet@fb.com>"
-  "Oleksiy Golovko <alexei.golovko@gmail.com>"
   "Rauan Mayemir <rauan@mayemir.io>"
-  "Carmelo Piccione <cep1@solvuu.com>"
-  "John Billings <john@monkeynut.org>"
   "Louis Roché <louis@louisroche.net>"
   "Brendan Long <self@brendanlong.com>"
-  "Chris Yocum <cyocum@gmail.com>"
-  "Louis Roché (Ahrefs) <louis.roche@ahrefs.com>"
-  "Louis Roché <louis.roche@ahrefs.com>"
-  "Pavel Antoshkin <pavel.antoshkin@ahrefs.com>"
-  "Pierre Boutillier <pierre.boutillier@laposte.net>"
-  "Shon Feder <shon.feder@key.me>"
-  "Anurag Soni <anuragsoni.13@gmail.com>"
-  "Arjun Ravi Narayan <arjunravinarayan@gmail.com>"
-  "Asya-kawai <kawai-toshiki@aintek.xyz>"
   "Christophe Troestler <christophe.Troestler@umons.ac.be>"
-  "Damien Doligez <ddoligez@janestreet.com>"
-  "Daniel M <dan.mntg@gmail.com>"
-  "Ding Xiang Fei <dingxiangfei2009@protonmail.ch>"
-  "François Pottier <francois.pottier@inria.fr>"
-  "Javier Chavarri <javier.chavarri@gmail.com>"
-  "Kate <kit.ty.kate@disroot.org>"
-  "Louis <louis.roche@ahrefs.com>"
-  "Louis Roché <louis@cryptosense.com>"
-  "Raman Varabets <raman+git@ahrefs.com>"
-  "Stephane Legrand <slegrand45@gmail.com>"
   "Vincent Bernardoff <vb@luminar.eu.org>"
   "haoyang <haoyang@esper.co>"
-  "pmundkur <prashanth.mundkur@gmail.com>"
-  "ygrek <ygrek@autistici.org>"
 ]
 license: "MIT"
 homepage: "https://github.com/ahrefs/atd"
 bug-reports: "https://github.com/ahrefs/atd/issues"
 depends: [
-  "dune" {>= "2.8"}
   "ocaml" {>= "4.08"}
+  "dune" {>= "2.8"}
   "atd" {>= "2.3.0"}
   "atdgen-runtime" {>= "2.0.0"}
   "atdgen-codec-runtime" {with-test}
@@ -95,3 +60,12 @@
     "@doc" {with-doc}
   ]
 ]
+url {
+  src:
+    "https://github.com/ahrefs/atd/releases/download/2.3.3/atdgen-codec-runtime-2.3.3.tbz"
+  checksum: [
+    "sha256=3a25addaebfe13b48a44fe3e6de88e9c50ea62433dfd3684f0a894d957d1eb47"
+    "sha512=a4375cb465b9d34b1bab49f331600b108241342e1957d9a2332eec5dd55ecf70021c21a6bb2431f015359e46dffbdae06a21c4f5c98c7e9ef3911bb7dca9461b"
+  ]
+}
+x-commit-hash: "28b612fbe750408ce58c3e952196460b2d7f5894"
--- atdgen-codec-runtime.opam	2022-03-17 18:15:35.081495794 -0700
+++ /home/martin/opam-repository/packages/atdgen-codec-runtime/atdgen-codec-runtime.2.3.3/opam	2022-03-16 21:30:51.075481705 -0700
@@ -1,4 +1,3 @@
-# This file is generated by dune, edit dune-project instead
 opam-version: "2.0"
 synopsis: "Runtime for atdgen generated bucklescript converters"
 description: """
@@ -11,53 +10,19 @@
 ]
 authors: [
   "Martin Jambon <martin@mjambon.com>"
-  "Rudi Grinberg <rudi.grinberg@gmail.com>"
-  "Martin Jambon <martin@r2c.dev>"
-  "Martin Jambon <github@mjambon.com>"
-  "Ivan Jager <aij+git@mrph.org>"
-  "oleksiy <oleksiy.golovko@ahrefs.com>"
   "David Sheets <sheets@alum.mit.edu>"
   "Rudi Grinberg <me@rgrinberg.com>"
-  "Martin Jambon <martin@esper.com>"
+  "Ivan Jager <aij+git@mrph.org>"
   "Jeff Meister <nanaki@gmail.com>"
-  "Caio Wakamatsu <caio.wakamatsu@ahrefs.com>"
   "Carmelo Piccione <carmelo.piccione@gmail.com>"
-  "Daniel Weil <danweil68@gmail.com>"
-  "Egor Chemokhonenko <egor.chemohonenko@ahrefs.com>"
-  "Gabriel Scherer <gabriel.scherer@gmail.com>"
   "Raman Varabets <roman.vorobets@gmail.com>"
-  "tzm <frank@boldsolutions.de>"
   "Mathieu Baudet <mathieubaudet@fb.com>"
-  "Oleksiy Golovko <alexei.golovko@gmail.com>"
   "Rauan Mayemir <rauan@mayemir.io>"
-  "Carmelo Piccione <cep1@solvuu.com>"
-  "John Billings <john@monkeynut.org>"
   "Louis Roché <louis@louisroche.net>"
   "Brendan Long <self@brendanlong.com>"
-  "Chris Yocum <cyocum@gmail.com>"
-  "Louis Roché (Ahrefs) <louis.roche@ahrefs.com>"
-  "Louis Roché <louis.roche@ahrefs.com>"
-  "Pavel Antoshkin <pavel.antoshkin@ahrefs.com>"
-  "Pierre Boutillier <pierre.boutillier@laposte.net>"
-  "Shon Feder <shon.feder@key.me>"
-  "Anurag Soni <anuragsoni.13@gmail.com>"
-  "Arjun Ravi Narayan <arjunravinarayan@gmail.com>"
-  "Asya-kawai <kawai-toshiki@aintek.xyz>"
   "Christophe Troestler <christophe.Troestler@umons.ac.be>"
-  "Damien Doligez <ddoligez@janestreet.com>"
-  "Daniel M <dan.mntg@gmail.com>"
-  "Ding Xiang Fei <dingxiangfei2009@protonmail.ch>"
-  "François Pottier <francois.pottier@inria.fr>"
-  "Javier Chavarri <javier.chavarri@gmail.com>"
-  "Kate <kit.ty.kate@disroot.org>"
-  "Louis <louis.roche@ahrefs.com>"
-  "Louis Roché <louis@cryptosense.com>"
-  "Raman Varabets <raman+git@ahrefs.com>"
-  "Stephane Legrand <slegrand45@gmail.com>"
   "Vincent Bernardoff <vb@luminar.eu.org>"
   "haoyang <haoyang@esper.co>"
-  "pmundkur <prashanth.mundkur@gmail.com>"
-  "ygrek <ygrek@autistici.org>"
 ]
 license: "MIT"
 homepage: "https://github.com/ahrefs/atd"
@@ -81,3 +46,12 @@
     "@doc" {with-doc}
   ]
 ]
+url {
+  src:
+    "https://github.com/ahrefs/atd/releases/download/2.3.3/atdgen-codec-runtime-2.3.3.tbz"
+  checksum: [
+    "sha256=3a25addaebfe13b48a44fe3e6de88e9c50ea62433dfd3684f0a894d957d1eb47"
+    "sha512=a4375cb465b9d34b1bab49f331600b108241342e1957d9a2332eec5dd55ecf70021c21a6bb2431f015359e46dffbdae06a21c4f5c98c7e9ef3911bb7dca9461b"
+  ]
+}
+x-commit-hash: "28b612fbe750408ce58c3e952196460b2d7f5894"
--- atdgen-runtime.opam	2022-03-17 18:15:35.081495794 -0700
+++ /home/martin/opam-repository/packages/atdgen-runtime/atdgen-runtime.2.3.3/opam	2022-03-16 21:30:51.075481705 -0700
@@ -1,4 +1,3 @@
-# This file is generated by dune, edit dune-project instead
 opam-version: "2.0"
 synopsis: "Runtime library for code generated by atdgen"
 description: """
@@ -11,60 +10,26 @@
 ]
 authors: [
   "Martin Jambon <martin@mjambon.com>"
-  "Rudi Grinberg <rudi.grinberg@gmail.com>"
-  "Martin Jambon <martin@r2c.dev>"
-  "Martin Jambon <github@mjambon.com>"
-  "Ivan Jager <aij+git@mrph.org>"
-  "oleksiy <oleksiy.golovko@ahrefs.com>"
   "David Sheets <sheets@alum.mit.edu>"
   "Rudi Grinberg <me@rgrinberg.com>"
-  "Martin Jambon <martin@esper.com>"
+  "Ivan Jager <aij+git@mrph.org>"
   "Jeff Meister <nanaki@gmail.com>"
-  "Caio Wakamatsu <caio.wakamatsu@ahrefs.com>"
   "Carmelo Piccione <carmelo.piccione@gmail.com>"
-  "Daniel Weil <danweil68@gmail.com>"
-  "Egor Chemokhonenko <egor.chemohonenko@ahrefs.com>"
-  "Gabriel Scherer <gabriel.scherer@gmail.com>"
   "Raman Varabets <roman.vorobets@gmail.com>"
-  "tzm <frank@boldsolutions.de>"
   "Mathieu Baudet <mathieubaudet@fb.com>"
-  "Oleksiy Golovko <alexei.golovko@gmail.com>"
   "Rauan Mayemir <rauan@mayemir.io>"
-  "Carmelo Piccione <cep1@solvuu.com>"
-  "John Billings <john@monkeynut.org>"
   "Louis Roché <louis@louisroche.net>"
   "Brendan Long <self@brendanlong.com>"
-  "Chris Yocum <cyocum@gmail.com>"
-  "Louis Roché (Ahrefs) <louis.roche@ahrefs.com>"
-  "Louis Roché <louis.roche@ahrefs.com>"
-  "Pavel Antoshkin <pavel.antoshkin@ahrefs.com>"
-  "Pierre Boutillier <pierre.boutillier@laposte.net>"
-  "Shon Feder <shon.feder@key.me>"
-  "Anurag Soni <anuragsoni.13@gmail.com>"
-  "Arjun Ravi Narayan <arjunravinarayan@gmail.com>"
-  "Asya-kawai <kawai-toshiki@aintek.xyz>"
   "Christophe Troestler <christophe.Troestler@umons.ac.be>"
-  "Damien Doligez <ddoligez@janestreet.com>"
-  "Daniel M <dan.mntg@gmail.com>"
-  "Ding Xiang Fei <dingxiangfei2009@protonmail.ch>"
-  "François Pottier <francois.pottier@inria.fr>"
-  "Javier Chavarri <javier.chavarri@gmail.com>"
-  "Kate <kit.ty.kate@disroot.org>"
-  "Louis <louis.roche@ahrefs.com>"
-  "Louis Roché <louis@cryptosense.com>"
-  "Raman Varabets <raman+git@ahrefs.com>"
-  "Stephane Legrand <slegrand45@gmail.com>"
   "Vincent Bernardoff <vb@luminar.eu.org>"
   "haoyang <haoyang@esper.co>"
-  "pmundkur <prashanth.mundkur@gmail.com>"
-  "ygrek <ygrek@autistici.org>"
 ]
 license: "MIT"
 homepage: "https://github.com/ahrefs/atd"
 bug-reports: "https://github.com/ahrefs/atd/issues"
 depends: [
-  "dune" {>= "2.8"}
   "ocaml" {>= "4.08"}
+  "dune" {>= "2.8"}
   "yojson" {>= "1.7.0"}
   "biniou" {>= "1.0.6"}
   "camlp-streams"
@@ -84,3 +49,12 @@
     "@doc" {with-doc}
   ]
 ]
+url {
+  src:
+    "https://github.com/ahrefs/atd/releases/download/2.3.3/atdgen-codec-runtime-2.3.3.tbz"
+  checksum: [
+    "sha256=3a25addaebfe13b48a44fe3e6de88e9c50ea62433dfd3684f0a894d957d1eb47"
+    "sha512=a4375cb465b9d34b1bab49f331600b108241342e1957d9a2332eec5dd55ecf70021c21a6bb2431f015359e46dffbdae06a21c4f5c98c7e9ef3911bb7dca9461b"
+  ]
+}
+x-commit-hash: "28b612fbe750408ce58c3e952196460b2d7f5894"
```

File: Makefile [hunk 1]
```diff
@@ -99,10 +101,22 @@ livedoc:
...
 opam-release:
-	dune-release distrib --skip-build --skip-lint --skip-tests -n $(package)
-	dune-release publish distrib --verbose -n $(package)
-	dune-release opam pkg -n $(package)
-	dune-release opam submit -n $(package)
+	dune-release tag
+	dune-release distrib
+	dune-release publish
+	dune-release opam pkg
+	dune-release opam submit
```
Comment (Khady) [inline Makefile:122]: this is the same as running `dune-release bistro` but maybe it's better to be explicit

File: Makefile [hunk 2]
```diff
@@ -99,10 +101,22 @@ livedoc:
...
 	python3 -m http.server 8888 --directory doc/_build
 
-package := atd
+# Prepare the opam files for a release. They're derived from 'dune-project'
+# by 'dune build' and then patched to work around some issues.
+# https://discuss.ocaml.org/t/when-exactly-does-dune-generate-opam-files/8292/2
+#
+.PHONY: opam-files
+opam-files:
+	$(DUNE) build
```
Comment (Khady) [inline Makefile:110]:
```suggestion
	$(DUNE) build *.opam
```
 would be a bit faster, there's no need to build all the code to generate the opam files. But that's nitpicking

**Review by Khady — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (mjambon) [reply Makefile:110]: that's great to know, thank you.

File: Makefile (hunk 1 above)
Comment (mjambon) [reply Makefile:122]: ah, I didn't know this. The help page assumes we'd figure this out.
```
       bistro
           For when you are in a hurry or need to go for a drink
```
well, I guess we can leave the makefile as it it. I'll probably change my mind if I have to run this a lot.

**Pull request merged**